packages/core/src/lib/auth0-repository.ts

```diff
@@ -0,0 +1,36 @@
+import { OidcUser } from "@dotkomonline/types"
+import { ManagementApiError, ManagementClient } from "auth0"
+
+export interface Auth0Repository {
+  getBySubject(sub: string): Promise<OidcUser | null>
+}
+
+export class Auth0RepositoryImpl implements Auth0Repository {
+  constructor(private readonly client: ManagementClient) {}
+  async getBySubject(sub: string) {
```

Comment:
If we have an access token we can just use `/userinfo`, no need to use the mgmt API.

What is in the `id_token` may have a better schema than the mgmt-api (e.g. `gender` is actually gender instead of being in `user_metadata/gender`, and we _might_ add stuff to the id_token that is not in auth0

Reply:
Probably for fetching attendees on an event?

Reply:
If that user has not logged in, there is no way for the data to go stale?

Reply:
What about this scenario:
1. User goes to their profile on our website and change their email
2. monoweb uses management api to update email in auth0

Now it is important that synchronizations after this actually use the new user state and not some stale data from id tokens. 

For the id tokens to contain the new user data, the user has to authorize again. Hitting the /userinfo endpoint will return an id token with stale data (old email before patch). Read [this](https://community.auth0.com/t/auth0-returns-a-stale-userinfo/13030/4)

If we just use the management api we can make sure we never use stale data from auth0 when synchronizing.

Thoughts?

Reply:
Ah, that is kinda unexpected but also understandable. Does the id_token update if you refresh the access_token?

Reply:
I don't know. I suggest we use the management api for now.